ShapesProject/Project1/TDynamicArray.h

```diff
@@ -0,0 +1,320 @@
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Reserve()
+	{
+		Resize(Capacity * 2);
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void ShrinkToFit()
+	{
+		Capacity = Size;
+		Resize(Capacity);
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Clear()
+	{
+		for (int i = 0; i < Size; i++)
+		{
+			delete DynamicArray[i];
+		}
+		Size = 0;
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void PushBack(const T& NewItem)
+	{
+		CheckBounds();
+		DynamicArray[Size] = NewItem;
+		Size++;
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void PopBack()
+	{
+		delete DynamicArray[Size];
+		Size--;
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Insert(const T& NewItem, const int Position)
+	{
+		CheckBounds();
+
+		Size++;
+
+		//Auxiliar variables
+		T AuxiliarA{ DynamicArray[Position] };
+		T AuxiliarB{ DynamicArray[Position] };
+
+		for (int i = Position; i < Size - 1; i++)
+		{
+			//moves the values +1 position
+			AuxiliarA = DynamicArray[i + 1];
+			DynamicArray[i + 1] = AuxiliarB;
+			AuxiliarB = AuxiliarA;
+		}
+		DynamicArray[Position] = NewItem;
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Erase(const int Position)
+	{
+		CheckBounds();
+
+		for (int i = Position; i < Size - 1; i++)
+		{
+			DynamicArray[i] = DynamicArray[i + 1];
+		}
+
+		delete DynamicArray[Size];
+		Size--;
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Fill(const T& Value)
+	{
+		for (int i = 0; i < Size; i++)
+		{
+			DynamicArray[i] = Value;
+		}
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	bool Swap(TDynamicArray& OtherArray)
+	{
+		int OtherSize = sizeof(OtherArray) / sizeof(OtherArray[0]);
+
+		if (OtherSize == Size)
+		{
+			T* AuxiliarArray = new T[Size];
+			memcpy(AuxiliarArray, DynamicArray, Size);
+			memcpy(DynamicArray, OtherArray, Size);
+			memcpy(OtherArray, AuxiliarArray, Size);
+
+			delete[] OtherArray;
+			return true;
+		}
+		return false;
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Append(const TDynamicArray& OtherArray)
+	{
+		int OtherArraySize = sizeof(OtherArray) / sizeof(T);
+		int NewSize = Size + OtherArraySize;
+		Resize(NewSize);
+
+		int OtherIndex = 0;
+
+		for (int i = Size; i < NewSize; i++)
+		{
+			DynamicArray[i] = OtherArray[OtherIndex];
+			OtherIndex++;
+		}
+
+		CheckBounds();
+	}
+
+	//------------------------------------------------------------------------------------------
+
+	void Resize(const int NewCapacity)
```

Comment:
Since the function is called Resize, it doesn't exactly make sense you're taking in a new capacity. I mean, it might modify the current capacity, but only because capacity depends on the size (in this case), so NewSize will make much more sense